02-microarray/clustering_microarray_01_heatmap.Rmd

```diff
@@ -1,145 +1,347 @@
 ## Create a heatmap
-To further customize the heatmap, see the vignettes on Bioconductor for a guide 
-at this [link](https://bioconductor.org/packages/devel/bioc/vignettes/ComplexHeatmap/inst/doc/s1.introduction.html).
-
-```{r Create a heatmap}
-# Reference the ComplexHeatmap guide for further customizing your heatmap
-browseVignettes("ComplexHeatmap")
-
-# Create an annotation that labels samples' groups by color
-annot <-
-  HeatmapAnnotation(df = data.frame(Groups = rep(c("SP", "MP"), # Replace with group names relevant to your data
-                                                 each = 2)),
-                    col = list(Groups = c("SP" = "green", "MP" = "orange"))) # Be sure to change the group names here as well -- a list of possible colors that can be used to replace the colors used here can be found at the following link: http://www.stat.columbia.edu/~tzheng/files/Rcolor.pdf
-
-# Create the heatmap object
-heatmap <- Heatmap(df.by.var, 
-        name = "Gene_Expression",
-        show_row_names = FALSE,
-        show_row_dend = FALSE,   # Can show the gene/row cluster if this is 
-        #changed to TRUE
-        column_dend_height = unit(4, "cm"),
-        bottom_annotation = annot) # assign the previously made annotation object
-
-# Print out the heatmap
-heatmap
+
+To further customize the heatmap, see a vignette for a guide at this [link](https://slowkow.com/notes/pheatmap-tutorial/) [@Slowikowski2017].
+
+```{r}
+# Create and store the heatmap object
+pheatmap <-
+  pheatmap(
+    df_by_var,
+    cluster_rows = TRUE, # We want to cluster the heatmap by rows (genes in this case)
+    cluster_cols = TRUE, # We also want to cluster the heatmap by columns (samples in this case),
+    show_rownames = FALSE, # We don't want to show the rownames because there are too many genes for the labels to be clearly seen
+    main = "Non-Annotated Heatmap",
+    colorRampPalette(c(
+      "deepskyblue",
+      "black",
+      "yellow"
+    ))(25
+    ),
+    scale = "row" # Scale values in the direction of genes (rows)
+  )
 ```
 
-## Save heatmap as a png
-You can easily switch this to save to a jpeg or tiff by changing the function 
-and file name within the function to the respective file suffix.
+We've created a heatmap but although our genes and samples are clustered, there is not much information that we can gather here because we did not provide the `pheatmap()` function with annotation labels for our samples.
+
+First let's save our clustered heatmap.
+
+### Save heatmap as a PNG
 
-```{r Save heatmap as a png}
+You can easily switch this to save to a jpeg or tiff by changing the function and file name within the function to the respective file suffix.
+
+```{r}
 # Open a png file
-png(file.path(plots_dir, 
-              "GSE12955_heatmap.png" # Replace file name with a relevant output plot name
-              ))
+png(file.path(
+  plots_dir,
+  "heatmap_GSE24862_non_annotated.png" # Replace file name with a relevant output plot name
```

Comment:
```suggestion
  "GSE24862_heatmap_non_annotated.png" # Replace file name with a relevant output plot name
```
Did I miss this elsewhere?

Reply:
This was just the format in the [`03-rnaseq/clustering_rnaseq_01_heatmap.Rmd`](https://github.com/AlexsLemonade/refinebio-examples/blob/e8500af03f3ab0fe25863f022d5fe44bd96b27c3/03-rnaseq/clustering_rnaseq_01_heatmap.Rmd#L323) notebook which I adapted some material from to get started on the notebook in this PR. 

I'll propagate these changes to that notebook in this PR as well.

Reply:
Gotcha. I might have missed that there.